cloud/lxd/lxd_container.py

```diff
@@ -0,0 +1,354 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2016, Hiroaki Nakamura <hnakamur@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+DOCUMENTATION = """
+---
+module: lxd_container
+short_description: Manage LXD Containers
+version_added: 2.2.0
+description:
+  - Management of LXD containers
+author: "Hiroaki Nakamura (@hnakamur)"
+options:
+    name:
+        description:
+          - Name of a container.
+        required: true
+    config:
+        description:
+          - A config dictionary for creating a container.
+            See https://github.com/lxc/lxd/blob/master/doc/rest-api.md#post-1
+          - Required when the container is not created yet and the state is
+            not absent.
+        required: false
+    state:
+        choices:
+          - started
+          - stopped
+          - restarted
+          - absent
+          - frozen
+        description:
+          - Define the state of a container.
+        required: false
+        default: started
+    timeout_for_addresses:
+        description:
+          - A timeout of waiting for IPv4 addresses are set to the all network
+            interfaces in the container after starting or restarting.
+          - If this value is equal to or less than 0, Ansible does not
+            wait for IPv4 addresses.
+        required: false
+        default: 0
+requirements:
+  - 'pylxd >= 2.0'
+notes:
+  - Containers must have a unique name. If you attempt to create a container
+    with a name that already existed in the users namespace the module will
+    simply return as "unchanged".
+"""
+
+EXAMPLES = """
+- name: Create a started container
+  lxd_container:
+    name: cent01
+    source: { type: image, alias: centos/7/amd64 }
```

Comment:
Should "source" here, and the other examples actually be "config"?

Can you please review the examples and ensure they work